### ISSUE TITLE
No way to identify "thread" when using pool=forks

### ISSUE THREAD

**Issue by alisd23** [vitest-dev/vitest#4982]
### Describe the bug

When using `pool=threads` the `VITEST_POOL_ID`, the reusable work thread being used can be identified, as it is always less than or equal to the max no. of workers (as stated here: https://vitest.dev/guide/migration.html#envs).

However this does not seem to be the case when `pool` is set to `threads`, the `VITEST_WORKER_ID` is the same, incrementing each time a new test file is run, however the `VITEST_POOL_ID` is not usable to share context across "threads" (forks) - as it seems to be some thread ID (e.g. 8603), for example when using a seeded database, one per fork.

Is the behaviour expected to be the same as with `pool=threads` or different?

### Reproduction

Any setup with `pool: 'threads'`- let me know if you need a reproduction but I assume you have many already in tests.

### System Info

```shell
Vitest: 1.2.0
Vite: 5.0.11

System:
    OS: macOS 14.1.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 4.26 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 4.0.2 - /usr/local/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
    pnpm: 8.14.0 - /usr/local/bin/pnpm
  Browsers:
    Chrome: 120.0.6099.216
    Safari: 17.1
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by AriPerkkio** [comment]
> When using `pool=threads` the VITEST_POOL_ID, the reusable work thread being used can be identified...
> However this does not seem to be the case when `pool` is set to `threads`

Just to be sure I understood correctly: Would you expect `VITEST_POOL_ID` to behave similarly with `pool=threads` and `pool=forks`?

This sounds like a bug in Tinypool that was introduced as early as in https://github.com/tinylibs/tinypool/pull/65. I thought Tinypool's `workerId` for threads was `node:worker_threads.threadId`, but it's actually some Tinypool's internal worker counter value. Right now `child_process` runtime (`pool=forks` in Vitest) is using `process.pid` there. It should be using Tinypool's `workerId` instead.

**Comment by alisd23** [comment]
> Just to be sure I understood correctly: Would you expect VITEST_POOL_ID to behave similarly with pool=threads and pool=forks?
I think that probably makes the most sense, unless you think otherwise. Another option could be a separate `VITEST_FORK_ID` or something, but the `VITEST_POOL_ID` should probably behave the same for all the options set by the `pool` config option.

Just so I understand then, the `child_process` runtime in tinypool needs to use the `workerId` somehow on [this line](https://github.com/tinylibs/tinypool/blob/ac2e83917557df2deff74b9b6ea0c83319f32a45/src/runtime/process-worker.ts#L28)?

**Comment by AriPerkkio** [comment]
It's this one: https://github.com/tinylibs/tinypool/blob/ac2e83917557df2deff74b9b6ea0c83319f32a45/src/entry/process.ts#L23.

Equivalent one on thread worker: https://github.com/tinylibs/tinypool/blob/ac2e83917557df2deff74b9b6ea0c83319f32a45/src/entry/worker.ts#L28.

**Comment by alisd23** [comment]
Ah I see (kinda) - thanks for the info. I'll have a go at fixing this on the tinypool side actually, hopefully not too difficult. Not sure if it'll be a breaking change or not, but will see what they say with a PR suggestion.

**Comment by AriPerkkio** [comment]
Awesome, thank you! I've created an issue on Tinypool's side and added some info where to start: https://github.com/tinylibs/tinypool/issues/80.

**Comment by AriPerkkio** [comment]
This is fixed in `tinypool@0.8.2`. You should be able to get that version running by updating your lockfile.
